app/services/setting_registry.rb

```diff
@@ -131,16 +131,29 @@ def load_definitions
     end
   end
 
-  def load_values
+  def known_categories
+    unless @known_descendants == Setting.descendants
+      @known_descendants = Setting.descendants
+      @known_categories = @known_descendants.map(&:name) << 'Setting'
+      @values_loaded = Date.new # force all values to be reloaded
+    end
+    @known_categories
+  end
+
+  def load_values(force: false)
     # we are loading only known STIs as we load settings fairly early the first time and plugin classes might not be loaded yet.
-    Setting.unscoped.where(category: Setting.descendants.map(&:name) + %w[Setting]).all.each do |s|
+    settings = Setting.unscoped.where(category: known_categories)
+    settings = settings.where('updated_at >= ?', @values_loaded) unless force
```

Comment:
This will always be the start of the application (the last time the `Setting.descendants` had changed), we might be more agressive and save the new timestamp on every values load.

Reply:
Never mind me, this is already there.